### PR TITLE
bots: Make atomic.install ready for Python 3

### PR DESCRIPTION
**Pull request by martinpitt**
This is the last thing in bots/ still requiring Python 2; unfortunately
{rhel,continuous}-atomic don't have Python 3 yet, so we can't actually
switch the interpreter yet.
    
Also add a vim modeline to force the file type to Python, as it
otherwise defaults to `sh` due to the `.install` file name.

### CODE REVIEW

File: bots/images/scripts/lib/atomic.install [hunk 1]
```diff
@@ -189,11 +191,11 @@ class AtomicCockpitInstaller:
 
     def package_basenames(self, package_names):
         """ convert a list of package names to a list of their basenames """
-        return filter(lambda s: not s is None, map(lambda s: self.package_basename(s), package_names))
+        return [s for s in [self.package_basename(s) for s in package_names] if s is not None]
```
Comment (stefwalter) [inline bots/images/scripts/lib/atomic.install]: Why is this change necessary. It's harder to read, no?

File: bots/images/scripts/lib/atomic.install [hunk 2]
```diff
@@ -225,7 +227,7 @@ class AtomicCockpitInstaller:
                     print("adding package %s (forced)" % (p))
                 packages_to_install.append(p)
 
-        packages_to_install = filter(lambda p: any(os.path.split(p)[1].startswith(base) for base in packages_to_install), self.rpms)
+        packages_to_install = [p for p in self.rpms if any(os.path.split(p)[1].startswith(base) for base in packages_to_install)]
```
Comment (stefwalter) [inline bots/images/scripts/lib/atomic.install]: Again, why is this change necessary?

File: bots/images/scripts/lib/atomic.install (hunk 1 above)
Comment (martinpitt) [reply bots/images/scripts/lib/atomic.install]: Interesting, I actually find the list comprehensions easier to read (and 2to3 suggests changing that); but I suppose that's a matter of what you are used to. In python3, `filter()` returns an iterator, not a list, so alternatively we can keep the filter/map and wrap the whole thing into a `list()`.

I did the latter now, and as a bonus dropped the unnecessary inner lambda.

**Comment by martinpitt** [comment]
@stefwalter: PTAL? Thanks!
